### PR TITLE
update event backend

### PR DESCRIPTION
**Pull request by kemalfidan**
This PR includes the backend code to update an event. It's pretty similar to `addEvent()` but I had to remove the existing event's images if they exist. Another thing that's important is updating all the `presentVolunteers'` hours if the event hours change. That's done in 1 trip to mongodb in a separate function under `action/Events.ts`. Also, I noticed the callback from fromidable's parse function wasn't catching any errors so I fixed that.

### CODE REVIEW

**Review by rossketron — APPROVED** [review]
Looks good to me!

File: server/actions/Event.ts [hunk 1]
```diff
@@ -12,8 +12,8 @@ export const getEvent = async function (id: string) {
         throw new APIError(400, "Invalid id");
     }
 
-    const event = (await EventSchema.findById(id)) as Event;
-    if (event == null) {
+    const event = await EventSchema.findById(id);
+    if (!event) {
```
Comment (andlrutt) [inline server/actions/Event.ts:16]: nice catch

File: server/actions/Event.ts [hunk 2]
```diff
@@ -46,6 +46,22 @@ export const addEvent = async function (event: Event) {
     await EventSchema.create(event);
 };
 
+/**
+ * @param id The _id of the event we want to update.
+ * @param event The nwe event object to insert into our database.
```
Comment (andlrutt) [inline server/actions/Event.ts]: small typo here!

File: server/actions/Volunteer.ts [hunk 3]
```diff
@@ -33,6 +33,16 @@ export const addVolunteer = async function (vol: Volunteer) {
     await VolunteerSchema.create(vol);
 };
 
+/**
+ * @param ids An array of volunteer objectIds whose hours need to be updated.
+ * @param ids The change that needs to be applied to each volunteer's hours. Can be negative.
```
Comment (andlrutt) [inline server/actions/Volunteer.ts]: should this be `@param change` instead of `@param ids`?

**Review by andlrutt — APPROVED** [review]
nice! I thought we were going to have to do a lot more to change the hours for all the attended volunteers if the event was edited, nice that it's that easy!

**Pull request merged**